src/components/molecules/TableComponent/TableComponent.tsx

```diff
@@ -72,92 +61,74 @@ export const TableComponent: React.FunctionComponent<TableComponentPropsType> =
-            alt={`${user.partyName || DEFAULT_PARTY_NAME} profile`}
-            width={imageSize}
-            height={imageSize}
-          />
-        );
-      }),
-    [imageSize, openUserProfileModal, users]
+      users.map((user) => (
+        <UserAvatar
+          key={user.id}
+          containerClassName={styles.userAvatar}
+          user={user}
+        />
+      )),
+    [users]
   );
 
   return (
-    <div className="TableComponent">
-      <div>
-        <div className="TableComponent__occupancy-warning">
+    <div className={styles.tableContainer}>
+      <div className={styles.tableHeader}>
+        {isRemoveButtonShown && (
+          <img
+            className={styles.deleteButton}
+            src={PortalCloseIcon}
+            alt="remove table"
+            onClick={toggleModal}
+          />
+        )}
+        <div>
           {locked && "locked"}
           {!locked && full && "full"}
         </div>
-        <div className="TableComponent__item-wrapper">
-          <span className="TableComponent__title">{table.title}</span>
-
-          {isRemoveButtonShown && (
-            <img
-              className="TableComponent__delete-icon"
-              src={PortalCloseIcon}
-              alt="remove table"
-              onClick={toggleModal}
-            />
-          )}
-        </div>
-        <div className="TableComponent__users">
-          {renderedUserPictures}
-
-          {users &&
-            table.capacity &&
-            table.capacity - users.length >= 0 &&
-            [...Array(table.capacity - users.length)].map((e, i) => (
-              <span
-                key={i}
-                onClick={() => onJoinClicked(table.reference, locked)}
-                id={`join-table-${venue?.name}-${table.reference}`}
-                className="TableComponent__add-user"
-              >
-                {STRING_PLUS}
-              </span>
-            ))}
-        </div>
+        <span>{table.title}</span>
       </div>
-      <Modal show={isModalShown} onHide={hideModal} centered bgVariant="dark">
-        <div className="TableComponent__modal-container">
-          <h2>Delete table</h2>
-          <p>
-            WARNING: This action cannot be undone and will permanently remove
-            {STRING_SPACE}
-            {table.title}
-          </p>
-          <div className="TableComponent__modal-buttons">
-            <ButtonNG
-              variant="secondary"
-              onClick={hideModal}
-              disabled={isDeletingTable}
-            >
-              Cancel
-            </ButtonNG>
-
-            <ButtonNG
-              disabled={isDeletingTable}
-              variant="danger"
-              onClick={removeTable}
-            >
-              Delete
-            </ButtonNG>
+      <div className={styles.tableUsers}>
+        {renderedUserPictures}
+
+        {users && table.capacity && table.capacity - users.length >= 0 && (
+          <span
+            key={"join-button"}
```

Comment:
[Q] Does this `key` prop perform any function at all outside a loop?

```suggestion
            key="join-button"
```

Reply:
Removed in 22c9bf0fd